includes/CLI/Plugin_Check_Command.php

```diff
@@ -0,0 +1,339 @@
+			$this->display_results( $formatter, $file_name, $file_results );
+		}
+	}
+
+	/**
+	 * Validate associative arguments.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $assoc_args List of the associative arguments.
+	 * @return array List of the associative arguments.
+	 *
+	 * @throws WP_CLI\ExitException Show error if plugin not found.
+	 */
+	protected function get_options( $assoc_args ) {
+
+		$options = array(
+			'checks'          => 'all',
+			'flag'            => 'stable',
+			'format'          => 'table',
+			'ignore-warnings' => false,
+			'ignore-errors'   => false,
+		);
+		$options = wp_parse_args( $assoc_args, $options );
+
+		if ( ! in_array( $options['flag'], $this->check_flags, true ) ) {
+
+			WP_CLI::error(
+				sprintf(
+					// translators: 1. Check flags.
+					__( 'Invalid flag argument, valid value will be one of [%1$s]', 'plugin-check' ),
+					implode( ', ', $this->check_flags )
+				)
+			);
+		}
+
+		if ( ! in_array( $options['format'], $this->output_formats, true ) ) {
+
+			WP_CLI::error(
+				sprintf(
+					// translators: 1. Output formats.
+					__( 'Invalid format argument, valid value will be one of [%1$s]', 'plugin-check' ),
+					implode( ', ', $this->output_formats )
+				)
+			);
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Gets the formatter instance to format check results.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $assoc_args Associative arguments.
+	 * @return WP_CLI\Formatter The formatter instance.
+	 */
+	protected function get_formatter( $assoc_args ) {
+
+		$default_fields = array(
+			'line',
+			'column',
+			'code',
+			'message',
+		);
+
+		if ( isset( $assoc_args['fields'] ) ) {
+			$default_fields = wp_parse_args( $assoc_args['fields'], $default_fields );
+		}
+
+		// If both errors and warnings are included, display the type of each result too.
+		if ( empty( $assoc_args['ignore_errors'] ) && empty( $assoc_args['ignore_warnings'] ) ) {
+			$default_fields = array(
+				'line',
+				'column',
+				'type',
+				'code',
+				'message',
+			);
+		}
+
+		return new WP_CLI\Formatter(
+			$assoc_args,
+			$default_fields
+		);
+	}
+
+	/**
+	 * Flattens and combines the given associative array of file errors and file warnings into a two-dimensional array.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $file_errors   Errors from a Check_Result, for a specific file.
+	 * @param array $file_warnings Warnings from a Check_Result, for a specific file.
+	 * @return array Combined file results.
+	 */
+	protected function flatten_file_results( $file_errors, $file_warnings ) {
+		$file_results = array();
+
+		foreach ( $file_errors as $line => $line_errors ) {
+			foreach ( $line_errors as $column => $column_errors ) {
+				foreach ( $column_errors as $column_error ) {
+
+					$file_results[] = array_merge(
+						$column_error,
+						array(
+							'type'   => 'ERROR',
+							'line'   => $line,
+							'column' => $column,
+						)
+					);
```

Comment:
Avoid using `array_merge` in loops because PHP will store a copy of the array in memory for each loop.
See: https://github.com/dseguy/clearPHP/blob/master/rules/no-array_merge-in-loop.md

In this case, I think you should be able to use the spread operator to unpack the `$column_error` value directly. Same applies to warnings below.


```suggestion
					$file_results[] = array(
						...$column_error,
						'type'   => 'ERROR',
						'line'   => $line,
						'column' => $column,
					);
```

Reply:
@joemcgill Yes, and this is a good point. This way works for us, but the spread operator with an associative array will not work for older than PHP 8.1 versions. Currently, the `Requires PHP is 5.6` in the plugin's doc comment. We might need to discuss this before implementing this change, but thank you for pointing this out. 🙌 

@joemcgill @jjgrainger @felixarntz Let me know what's your thoughts here. Thanks! 🙂

Reply:
You're right @vishalkakadiya, I was conflating argument unpacking with array unpacking in my example. On a second look, the way you're using `array_merge` here isn't too bad, considering that you're not merging with an array that grows overtime. There are probably optimizations to be made here, but this is likely fine for now.